src/Persistence/Sql/Sqlite/CreateRegexpLikeFunctionMiddleware.php

```diff
@@ -30,18 +30,13 @@ public function connect(
                         return null;
                     }
 
-                    if (is_int($value)) {
-                        $value = (string) $value;
-                    } elseif (is_float($value)) {
-                        $value = Expression::castFloatToString($value);
-                    }
+                    $value = CreateRegexpLikeFunctionMiddleware::castScalarToString($value);
 
-                    $isValidUtf8Value = \PHP_VERSION_ID < 80200
-                        ? preg_match('~~u', $value) === 1 // much faster in PHP 8.1 and lower
```

Comment:
is this performance flaw fixed in newer PHP 8.2, 8.3 versions?

Reply:
It is/was a hack I invented for atk4/ui, but here we use it less often. I will recheck the performance here with some real data...

Reply:
should be fine - https://3v4l.org/aO7XS - 100k strings take ~20ms

wrapping the check in a function with both variants would imply function call overhead - https://3v4l.org/tanVc

Reply:
should be fine - https://3v4l.org/QaZpS - 100k strings take ~20ms

but I have landed the original speedup hack - https://github.com/atk4/data/pull/1203/commits/b9e189d7ee1389db18c16e92bcc9b04459eb82df